test/e2e/specs/javascript.spec.js

```diff
@@ -1,14 +1,121 @@
+      .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+      .end();
+  },
+
+  'Test processor-javascript' : function (client) {
+    client
+      .url(client.launch_url)
+      .selectTab('JavaScript')
+      .selectJsProcessor("javascript")
+      .setJsValue("document.body.innerHTML = 'Hello'")
+      .selectOutputFrame()
+      .assert.containsText('body', 'Hello')
+      .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+      .end();
+  },
+
+  'Test processor-babel' : function (client) {
+    client
+        .url(client.launch_url)
+        .selectTab('JavaScript')
+        .selectJsProcessor("babel")
+        .setJsValue("const value = 'Hello';\ndocument.body.innerHTML = value;")
+        .selectOutputFrame()
+        .assert.containsText('body', 'Hello')
+        .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+        .end();
+  },
+
+  'Test processor-jsx' : function (client) {
+    client
+        .url(client.launch_url)
+        .setHtml("<div id='root'> </div>")
+        .selectTab('JavaScript')
+        .selectJsProcessor("jsx")
+        .setJsValue("ReactDOM.render\n(<div>Hello</div>,\n document.getElementById('root'));")
+        .selectOutputFrame()
+        .assert.containsText('body', 'Hello')
+        .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+        .end();
+  },
+
+  'Test processor-coffeescript' : function (client) {
+    client
+          .url(client.launch_url)
+          .selectTab('JavaScript')
+          .selectJsProcessor("coffeescript")
+          .setJsValue("class Foo\n bar: 'Hello'\nobj = new Foo()\ndocument.body.innerHTML = obj.bar")
+          .selectOutputFrame()
+          .assert.containsText('body', 'Hello')
+          .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+          .end();
+  },
+
+  'Test processor-traceur' : function (client) {
+    client
+      .url(client.launch_url)
+      .selectTab('JavaScript')
+      .selectJsProcessor("traceur")
+      .setJsValue("const value = 'Hello';\ndocument.body.innerHTML = value;")
+      .selectOutputFrame()
+      .assert.containsText('body', 'Hello')
+      .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+      .end();
+  },
+
+  'Test processor-typescript' : function (client) {
+    client
+      .url(client.launch_url)
+      .selectTab('JavaScript')
+      .selectJsProcessor("typescript")
+      .setJsValue("let value: string = 'Hello';\ndocument.body.innerHTML = value;")
+      .selectOutputFrame()
+      .assert.containsText('body', 'Hello')
+      .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
+      .end();
+  },
+
+  'Test processor-processing' : function (client) {
+    client
+      .url(client.launch_url)
+      .selectTab('JavaScript')
+      .selectJsProcessor("processing")
+      .setJsValue("size(200, 200);\nbackground(204);")
+      .selectOutputFrame()
+      .assert.attributeEquals("canvas", "width", "200")
+      .assert.attributeEquals("canvas", "height", "200")
+      .assert.urlMatch(/\/\w+\/edit\?html,js,output$/)
```

Comment:
I don't think you need to test URL much everywhere, just the first one should be enough